code/object/objectsnd.cpp

```diff
@@ -391,18 +391,24 @@ void maybe_play_flyby_snd(float closest_dist, object *closest_objp, object *list
 					return;
 				}
 
-				// pick a random species-based sound
+				// check to see if a flyby sound is specified in ship class
+				// if not, then pick a random species-based sound
 
-				ship_info *sip = &Ship_info[Ships[closest_objp->instance].ship_info_index];
+				int shipinfo_int = Ships[closest_objp->instance].ship_info_index;
+				ship_info* sip = &Ship_info[shipinfo_int];
 				game_snd *snd;
 
-				if (sip->flags[Ship::Info_Flags::Bomber])
-					snd = &Species_info[sip->species].snd_flyby_bomber;
-				else
-					snd = &Species_info[sip->species].snd_flyby_fighter;
+				if (Ship_info[shipinfo_int].glide_end_snd.isValid()) {
```

Comment:
This can be expressed much more succinctly as `sip->glide_end_snd.isValid()`, skipping the extra index variable... except you don't want `glide_end_snd`, you want the new `flyby_snd`.